### ISSUE TITLE
throws incorrect error on properly implemented route given unimplemented root route

### ISSUE THREAD

**Issue by ashleygwilliams** [chrisdickinson/reverse#2]
ok instead of a full example i opted for a test. see #3

given:
```
const controller = {
  greet () {
  }
}

const router = reverse`
  GET /           hello
  GET /greetings  greet
`(controller)
```

we expect:

- `/greetings` => OK

we get:

- `/greetings` => `not ok expected controller to provide hello`

<hr/>

it is worth noting that:

given:
```
const controller = {
  greet () {
  }
}

const router = reverse`
  GET /hello      hello
  GET /greetings  greet
`(controller)
```

we get what we expect:

- `/greetings` => OK

**Comment by ashleygwilliams** [comment]
i think it's worth noting that this error message is not *exactly* wrong, just a bit irrelevant to the requested route. fixing this may also involve fixing error messaging in knork:

- code https://github.com/npm/knork/blob/cc49ced8d433024fa4a20ac042f250b573e6c47e/knork.js#L143-L145
- test https://github.com/npm/knork/blob/cc49ced8d433024fa4a20ac042f250b573e6c47e/test/server-test.js#L409

**Comment by ashleygwilliams** [comment]
there is an assumption here that the router should work for routes that are implemented even if some others are not. i think this is a reasonable assumption to have, but would be willing to understand if that's not an assumption we want to have.

**Comment by chrisdickinson** [comment]
Fixed by #3!
